compiler/rustc_hir/src/hir.rs

```diff
@@ -859,7 +859,11 @@ impl fmt::Debug for OwnerNodes<'_> {
                 &self
                     .nodes
                     .iter_enumerated()
-                    .map(|(id, parented_node)| (id, parented_node.as_ref().map(|node| node.parent)))
+                    .map(|(id, parented_node)| {
+                        let parented_node = parented_node.as_ref().map(|node| node.parent);
+
+                        debug_fn(move |f| write!(f, "({id:?}, {parented_node:?})"))
```

Comment:
Would `format_args!()` work?

Reply:
It woudn't, as `id` is an owned value which only lives as long as the closure (unless we move it, as we do here). Unless we can move a value into `format_args` somehow.

Reply:
Right